core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -420,12 +502,68 @@ public LoggingDecoratorBuilder responseCauseSanitizer(
         return responseCauseSanitizer;
     }
 
+    /**
+     * Sets the {@link Predicate} for evaluating to log response cause or not.
+     * You can prevent from logging the response cause by returning {@code true}
+     * in the {@link Predicate}. If unset, will log a response cause.
+     */
+    @UnstableApi
+    public LoggingDecoratorBuilder responseCauseFilter(Predicate<Throwable> responseCauseFilter) {
+        this.responseCauseFilter = requireNonNull(responseCauseFilter, "responseCauseFilter");
+        return this;
+    }
+
+    /**
+     * Returns the {@link Predicate} to evaluate if logging response cause or not.
+     */
+    @UnstableApi
+    protected final Predicate<Throwable> responseCauseFilter() {
+        return responseCauseFilter;
+    }
+
+    /**
+     * Sets the {@link LogFormatter} to converts a {@link RequestOnlyLog} or {@link RequestLog}
+     * into log message. If unset, will use {@link TextLogFormatter}.
+     *
+     * @throws IllegalStateException If both the log sanitizers and the {@link LogFormatter} are specified.
+     */
+    @UnstableApi
+    public LoggingDecoratorBuilder logFormatter(LogFormatter logFormatter) {
+        this.logFormatter = requireNonNull(logFormatter, "logFormatter");
+        if (useSanitizers) {
+            throw new IllegalStateException(
+                    "The log sanitizers and the LogFormatter cannot be used at the same time");
+        }
+        return this;
+    }
+
+    /**
+     * Builds the {@link LogFormatter} to converts a {@link RequestOnlyLog} or {@link RequestLog}
+     * into log message. If {@link LogFormatter} is set, returns it.
+     * If not set, returns {@link TextLogFormatter} that is built with the sanitizers in this builder.
```

Comment:
slight rewording 😅  (also `TextLogFormatter` is package private)

```suggestion
     * Returns a {@link LogFormatter} which converts a {@link RequestOnlyLog} or {@link RequestLog}
     * into a log message. If a {@link LogFormatter} has been set using {@link #logFormatter}, it is returned.
     * Otherwise, a text based {@link LogFormatter} built using the set sanitizers is returned.
```